analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -147,4 +147,23 @@ public override void ExecutionCompleted()
             ReportIssue(operation, operation.Syntax.ToString());
         }
     }
+
+    private static ProgramState SetConstraintForObjectCreation(ProgramState state, IObjectCreationOperationWrapper objectCreation) =>
+        FirstArgumentAsCollection(objectCreation) switch
+        {
+            { } firstArgument when ConstraintFromArgument(state, firstArgument) is { } constraint => state.SetOperationConstraint(objectCreation, constraint),
```

Comment:
It doens't need to be the 1st one when calling stuff out of order.
```suggestion
            { } collectionArgument when ConstraintFromArgument(state, collectionArgument) is { } constraint => state.SetOperationConstraint(objectCreation, constraint),
```